doc/style_guide.md

```diff
@@ -295,6 +295,12 @@ Code style rules
 - Use `TryGetValue` instead of first calling `Dictionary.ContainsKey`
   and then reading the value separate because `TryGetValue` is faster.
 
+- When trying to save Thrive-related objects as a member variable of a
+  class/interface (especially scene-based nodes) such as Microbe,
+  FloatingChunks, etc. you should put `[UseThriveSerializer]` attribute on
+  top of the class/interface so that they can be recognized by the Thrive
+  serializer and thus be deserialized accordingly.
```

Comment:
I'm a bit suspicious about this advice. Especially as the scene loaded attribute is much more important to make objects that are created from scenes work. Also this doesn't clarify *why* it makes them work. The *only* thing really the Thrive serializer does is that it has slightly different order it deserializes the reference to self and child properties. And of course it supports the additional attributes, which may have been what you wanted? But I'm unsure, so please clarify.

Reply:
Yeah I might have not explained it accurately here. I think from your explanation previously why this was needed was that the `UseThriveSerializer` needs to be specified for the base type if it dynamically will deserialize a type that needs the thrive serializer, more or less.

Reply:
That doesn't sound entirely right... What's the concrete place where you encountered the issue? It can't be that child properties cannot use a different serializer than the parent, because obviously we have places where that works.

Reply:
Improved documentation.